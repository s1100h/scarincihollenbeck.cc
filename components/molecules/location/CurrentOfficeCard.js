import { MdLocalPhone } from 'react-icons/md';
import { IoPrintSharp } from 'react-icons/io5';
import { FaLocationDot } from 'react-icons/fa6';
import { OfficeCardContainer } from '../../../styles/Locations.style';

const CurrentOfficeCard = ({
  children,
  phone,
  fax,
  streetAddress,
  floor,
  postCode,
  addressLocality,
  addressRegion,
  backgroundColor,
}) => (
  <OfficeCardContainer backgroundColor={backgroundColor}>
    {children}
    <address>
      <span>
        <span>
          <FaLocationDot size={18} />
        </span>
        {streetAddress}
        {streetAddress?.length > 0 && ', '}
        {floor}
        {floor?.length > 0 && ', '}
        {addressLocality}
        {addressLocality?.length > 0 && ', '}
        {`${addressRegion} ${postCode}`}
      </span>
      <a href={`tel:${phone}`}>
        <MdLocalPhone size={20} />
        {phone}
      </a>
      <span>
        <IoPrintSharp size={20} />
        {fax}
      </span>
    </address>
  </OfficeCardContainer>
);

export default CurrentOfficeCard;
