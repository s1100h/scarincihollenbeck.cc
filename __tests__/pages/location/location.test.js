import CurrentOfficeCard from 'components/molecules/location/CurrentOfficeCard';
import renderer from 'react-test-renderer';

describe('Location page', () => {
  it('Contacts snapshot', async () => {
    const props = {
      phone: '212-286-0747',
      fax: '212-808-4155',
      streetAddress: '519 8th Avenue',
      floor: '25th Floor',
      addressRegion: 'NY',
      postCode: '10018',
      addressLocality: 'New York',
    };

    const tree = renderer
      .create(
        <CurrentOfficeCard {...props}>
          <h3 className="current-office-card-title">New York City</h3>
        </CurrentOfficeCard>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
