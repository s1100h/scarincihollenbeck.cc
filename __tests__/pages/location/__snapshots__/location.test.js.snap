// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location page Contacts snapshot 1`] = `
<div
  className="sc-929077e1-7 guZbsx"
>
  <h3
    className="current-office-card-title"
  >
    New York City
  </h3>
  <address>
    <span>
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16s6-5.686 6-10A6 6 0 0 0 2 6c0 4.314 6 10 6 10zm0-7a3 3 0 1 1 0-6 3 3 0 0 1 0 6z"
          />
        </svg>
      </span>
      519 8th Avenue
      , 
      25th Floor
      , 
      New York
      , 
      NY 10018
    </span>
    <a
      href="tel:212-286-0747"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
          fillRule="evenodd"
        />
      </svg>
      212-286-0747
    </a>
    <a
      href="fax:212-808-4155"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 1a2 2 0 0 0-2 2v1h10V3a2 2 0 0 0-2-2H5zm6 8H5a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-3a1 1 0 0 0-1-1z"
        />
        <path
          d="M0 7a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2h-1v-2a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v2H2a2 2 0 0 1-2-2V7zm2.5 1a.5.5 0 1 0 0-1 .5.5 0 0 0 0 1z"
        />
      </svg>
      212-808-4155
    </a>
  </address>
</div>
`;
